design/routing-design.md

```diff
@@ -123,6 +123,34 @@ Exact Path based routing makes decisions based upon the path of the request (e.g
 Wildcard path based routing allows for a portion of the path to be dynamic (e.g. /path/*/foo).
 This is useful to allow for many types of request paths to be routed to the same backed without specifying them explicitly up front.
 
+- Wildcards defined cannot be the end of the path (e.g. `/foo/*/bar` or `/*/foo` not `/foo*`)
+- Similar to path prefix matching, wildcard path routing requires the condition to start with a `/`
+- There is no change to the sorting of paths. The path `/api/*/foo` would still sort before prefix: `/api/v2`
+- Merging conditions together is allowed. If users define a prefix condition of `/foo` and include that to another Proxy which defines a wildcard path of `/*/bar`, the result is a regex expression of `/foo/*/bar`
+- If the wildcard path match is defined in a `prefix` condition, then the resulting match follows a prefix style match (Note: Exact path routing isn't yet enabled)
+- Wildcard path matches cannot be defined inside an `Include` (see next section)
```

Comment:
I'm not sure if this is too restrictive, but it would be difficult to detect these types of conflicts in the DAG without doing a 2nd pass over it to validate. 

My problem isn't as much detecting, but understanding who is authoritative or not when the conflict arises which is why I thought of going this path first.

Reply:
I agree with you that restricting wildcards to just routes is probably a good first step, it'll always be possible to lift that restriction in the future.